conda_env/cli/main_create.py

```diff
@@ -33,54 +33,63 @@ def configure_parser(sub_parsers):
         help=description,
         epilog=example,
     )
-
     p.add_argument(
-        '-n', '--name',
+        '-f', '--file',
         action='store',
-        help='name of environment (in %s)' % os.pathsep.join(config.envs_dirs),
-        default=None,
+        help='environment definition file (default: environment.yml)',
+        default='environment.yml',
     )
-
     p.add_argument(
-        '-f', '--file',
+        '-n', '--name',
         action='store',
-        help='environment definition (default: environment.yml)',
-        default='environment.yml',
+        help='environment definition [Deprecated]',
+        default=None,
+        dest='old_name'
     )
     p.add_argument(
         '-q', '--quiet',
+        action='store_false',
         default=False,
     )
+    p.add_argument(
+        'name',
+        help='environment definition',
+        action='store',
+        default=None,
+        nargs='?'
+    )
     common.add_parser_json(p)
     p.set_defaults(func=execute)
 
 
 def execute(args, parser):
-    try:
-        env = from_file(args.file)
-    except exceptions.EnvironmentFileNotFound as e:
-        msg = 'Unable to locate environment file: %s\n\n' % e.filename
-        msg += "\n".join(textwrap.wrap(textwrap.dedent("""
-            Please verify that the above file is present and that you have
-            permission read the file's contents.  Note, you can specify the
-            file to use by explictly adding --file=/path/to/file when calling
-            conda env create.""").lstrip()))
-
-        common.error_and_exit(msg, json=args.json)
-
-    if not args.name:
-        if not env.name:
-            # TODO It would be nice to be able to format this more cleanly
-            common.error_and_exit(
-                'An environment name is required.\n\n'
-                'You can either specify one directly with --name or you can add\n'
-                'a name property to your %s file.' % args.file,
-                json=args.json
-            )
-        # Note: stubbing out the args object as all of the
-        # conda.cli.common code thinks that name will always
-        # be specified.
-        args.name = env.name
+
+    if args.name or args.old_name:
+        if args.name is not None:
+            name = args.name
+        else:
+            name = args.old_name
+            print("`--name` is deprecated. Use:\n"
+                  "  conda env create {}".format(args.old_name))
+
+        for Spec in all_specs:
+            spec = Spec(name)
+            if spec.can_process():
+                env = from_yaml(spec.environment)
+                args.name = env.name
+
+    else:
```

Comment:
The way this `if/else` is setup it changes the behavior subtly.  It used to be that `--name` was an override to anything that was found in the `--file` (implied or explicit), now you either use a name or a file, but both can't be used together.  I'm going to open up a PR against this with some tweaks.